files/class-wp-filesystem-vip.php

```diff
@@ -68,10 +68,11 @@ private function is_tmp_path( $file_path ) {
 	}
 
 	private function is_uploads_path( $file_path ) {
-		$upload_dir  = wp_get_upload_dir();
-		$upload_base = $upload_dir['basedir'];
+		// This probably needs to be more robust as I'm not sure if there are ever scenarios where the uploads directory would _not_ be /tmp/uploads on VIP Go
+		// We just need to look for /tmp/uploads here - we don't really care about the /sites/{site_id} part to determine if it's an uploads path or not
+		$upload_path = trailingslashit( get_temp_dir() ) . 'uploads';
```

Comment:
I am a little concerned with this since it makes it less flexible with the assumption that the uploads directory will always be in `/tmp`. Especially in the context of the Filesystem Stream Wrapper work I'm doing.

Reply:
> I am a little concerned with this

Same. I'm not really sure how to use something more dynamic, though, unless we can implement a method that calculates where the uploads directory _should_ be and exclude the `/sites/{site_id}` portion. The reason is that the purpose of the `is_uploads_path()` method is to determine the transport needed to manipulate a given file path - `WP_Filesystem_VIP_Uploads` if it's in the "uploads" directory, or `WP_Filesystem_Direct` if it isn't. The problem is that if we use `wp_get_upload_dir()['basedir']` to check, paths that are _actually_ in the uploads directory and _need_ `WP_Filesystem_VIP_Uploads` to be manipulated will fail this check. Example: You're working on site ID 3 and want to duplicate it to a new subsite. Trying to copy media files from site ID 3 to site ID 7 would fail because `wp_get_upload_dir()['basedir']` is `/tmp/uploads/sites/3` and it's trying to determine the transport for a path (`/tmp/uploads/sites/7/some-file.jpg`) that doesn't include `/tmp/uploads/sites/3`. So the existing logic prevents you from being able to write to any other site on the multisite which is a blocker for some of our clients. Here we need a reliable way to determine whether or not `WP_Filesystem_VIP_Uploads` should be used to deal with the file path given.

Do you know how the `/tmp/uploads` path is determined to be the "uploads" directory? If it's hard coded somewhere, then I think we're fine to make this assumption but if it can be changed by clients we'll need a way to get the value they've set it to.

> Especially in the context of the Filesystem Stream Wrapper work I'm doing

Would this mean that `WP_Filesystem_VIP_Uploads` may need to be tweaked to account for that? A way for it to determine whether or not it's needed or if it can just use `WP_Filesystem_Direct`.

Reply:
> Trying to copy media files from site ID 3 to site ID 7 would fail because wp_get_upload_dir()['basedir'] is /tmp/uploads/sites/3 and it's trying to determine the transport for a path (/tmp/uploads/sites/7/some-file.jpg) that doesn't include /tmp/uploads/sites/3

Perhaps your change to `get_upload_dir` shouldn't change how `basedir` is set? As per your example, `path` can remain as `/tmp/uploads/sites/3/2018/12/` but `basedir` is simply `/tmp/uploads/`. Would that resolve this issue?

> Do you know how the /tmp/uploads path is determined to be the "uploads" directory? If it's hard coded somewhere, then I think we're fine to make this assumption but if it can be changed by clients we'll need a way to get the value they've set it to.

It is set as a constant, `LOCAL_UPLOADS` in `a8c-files.php`. So it is hard coded but I still feel that it's better to stick to using `wp_get_upload_dir` to get the upload paths since any changes made to the `LOCAL_UPLOADS` would be automatically reflected by that function. In the StreamWrapper plugin I've been working on, we're hooking into a filter to change the output of `wp_get_upload_dir` so there's that situation too.

> Would this mean that WP_Filesystem_VIP_Uploads may need to be tweaked to account for that?

Actually, from our tests so far, it looks like the `WP_Filesystem_VIP_Uploads` class is not even needed.